src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyPropertyGenerator.cs

```diff
@@ -0,0 +1,463 @@
+						{
+							builder.AppendLineInvariant($"backingFieldOwnerInstance.{backingFieldName} = ({propertyTypeName})instance.GetValue({propertyOwnerTypeName}.{propertyName}Property);");
+							builder.AppendLineInvariant($"backingFieldOwnerInstance.{backingFieldName}Set = true;");
+						}
+					}
+				}
+
+				if (coerceCallback || propertyOwnerType.GetMethods().Any(m => m.Name == "Coerce" + propertyName))
+				{
+					builder.AppendLineInvariant($"\t\t, coerceValueCallback: (instance, baseValue) => Coerce{propertyName}(instance, ({propertyTypeName})baseValue)");
+				}
+
+				changedCallbackName ??= $"On{propertyName}Changed";
+				var propertyChangedMethod = propertyOwnerType.GetMethods().FirstOrDefault(m => m.Name == changedCallbackName);
+				if (changedCallback || propertyChangedMethod != null)
+				{
+					var isDPChangedEventArgsParam = Equals(propertyChangedMethod?.Parameters.ElementAtOrDefault(1)?.Type, _dependencyPropertyChangedEventArgsSymbol);
+					if (isDPChangedEventArgsParam)
+					{
+						builder.AppendLineInvariant($"\t\t, propertyChangedCallback: (instance, args) => {changedCallbackName}(instance, args)");
+					}
+					else
+					{
+						builder.AppendLineInvariant($"\t\t, propertyChangedCallback: (instance, args) => {changedCallbackName}(instance, ({propertyTypeName})args.OldValue, ({propertyTypeName})args.NewValue)");
+					}
+				}
+
+
+				builder.AppendLineInvariant($"));");
+
+				builder.AppendLineInvariant($"#endregion");
+			}
+
+			static KeyValuePair<string, TypedConstant>? GetAttributeValue(AttributeData attribute, string parameterName)
+				=> attribute?.NamedArguments.FirstOrDefault(kvp => kvp.Key == parameterName);
+
+			static bool GetBooleanAttributeValue(AttributeData attribute, string parameterName, bool defaultValue = true)
+				=> attribute?.NamedArguments.FirstOrDefault(kvp => kvp.Key == parameterName).Value.Value is bool value ? value : defaultValue;
+
+			private void GenerateProperty(IndentedStringBuilder builder, INamedTypeSymbol typeSymbol, IPropertySymbol propertySymbol, AttributeData attribute)
+			{
+				var propertyName = propertySymbol.Name;
+				var propertyTypeName = propertySymbol.Type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+				var containingTypeName = propertySymbol.ContainingType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+				var changedCallbackName = GetAttributeValue(attribute, "ChangedCallbackName")?.Value.Value?.ToString();
+				var metadataOptions = GetAttributeValue(attribute, "Options")?.Value.Value?.ToString() ?? "0";
+				var coerceCallback = GetBooleanAttributeValue(attribute, "CoerceCallback", false);
+				var changedCallback = GetBooleanAttributeValue(attribute, "ChangedCallback", false);
+				var generateAsField = GetBooleanAttributeValue(attribute, "GenerateAsField", true);
+				var localCache = GetBooleanAttributeValue(attribute, "LocalCache", true);
+				var isPublic = GetBooleanAttributeValue(attribute, "IsPublic", true);
```

Comment:
Codacy found an issue: [Remove this default value assigned to parameter 'defaultValue'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5714227)